modin/core/dataframe/pandas/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,467 @@
+
+        new_partitions = self._col._partition_mgr_cls.map_axis_partitions(
+            0,
+            self._col._partitions,
+            lambda df: df,
+            keep_partitioning=False,
+            lengths=new_lengths,
+        )
+        new_df = self._col.__constructor__(
+            new_partitions,
+            self._col.index,
+            self._col.columns,
+            new_lengths,
+            self._col._column_widths,
+        )
+        for length in new_df._row_lengths:
+            yield PandasProtocolColumn(
+                self._col.mask(row_positions=range(length), col_positions=None),
+                allow_copy=self._allow_copy,
+                offset=offset,
+            )
+            offset += length
+
+    def get_buffers(self) -> Dict[str, Any]:
+        buffers = {}
+        buffers["data"] = self._get_data_buffer()
+        try:
+            buffers["validity"] = self._get_validity_buffer()
+        except Exception:
+            buffers["validity"] = None
+
+        try:
+            buffers["offsets"] = self._get_offsets_buffer()
+        except Exception:
+            buffers["offsets"] = None
+
+        return buffers
+
+    def _get_data_buffer(
+        self,
+    ) -> Tuple[PandasProtocolBuffer, Any]:  # Any is for self.dtype tuple
+        """
+        Return the buffer containing the data and the buffer's associated dtype.
+
+        Returns
+        -------
+        tuple
+            The data buffer.
+        """
+        _k = DTypeKind
+        dtype = self.dtype
+        if dtype[0] in (_k.INT, _k.UINT, _k.FLOAT, _k.BOOL):
+            buffer = PandasProtocolBuffer(
+                self._col.to_numpy().flatten(), allow_copy=self._allow_copy
+            )
+        elif dtype[0] == _k.CATEGORICAL:
+            pandas_series = self._col.to_pandas().squeeze(axis=1)
+            codes = pandas_series.values.codes
+            buffer = PandasProtocolBuffer(codes, allow_copy=self._allow_copy)
+            dtype = self._dtype_from_primitive_pandas_dtype(codes.dtype)
+        elif dtype[0] == _k.STRING:
+            # Marshal the strings from a NumPy object array into a byte array
+            buf = self._col.to_numpy().flatten()
+            b = bytearray()
+
+            # TODO: this for-loop is slow; can be implemented in Cython/C/C++ later
+            for i in range(buf.size):
+                if type(buf[i]) == str:
+                    b.extend(buf[i].encode(encoding="utf-8"))
+
+            # Convert the byte array to a pandas "buffer" using a NumPy array as the backing store
+            buffer = PandasProtocolBuffer(np.frombuffer(b, dtype="uint8"))
+
+            # Define the dtype for the returned buffer
+            dtype = (
+                _k.STRING,
+                8,
+                "u",
+                "=",
+            )  # note: currently only support native endianness
+        else:
+            raise NotImplementedError(f"Data type {self._col.dtype[0]} not handled yet")
```

Comment:
Another drawback of not using a specific exception type [here](https://github.com/modin-project/modin/pull/4150#discussion_r811878525). We don't want to return a None buffer if we have not implemented something yet, we have to raise an explicit error.

Reply:
fixed